authnz/src/main/java/io/stargate/auth/entity/AuthorizedResource.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.auth.entity;
+
+import org.immutables.value.Value;
+
+/** Identifies a resource for authorization purposes. */
+@Value.Immutable
+public interface AuthorizedResource<W> extends AuthEntityWrapper<W> {
+
+  /** Identifies the resource kind. */
+  @Value.Parameter
+  ResourceKind kind();
+
+  /**
+   * Identifies the scope of the resource according to the resource {@link #kind}.
+   *
+   * <p>For roles the scope should be {@code null}.
+   *
+   * <p>For tables and functions the scope is the keyspace name. If {@code null}, it means "all
+   * keyspaces" (in which case {@link #element()} should be {@code null} too).
+   */
+  @Value.Default
+  default String scope() {
+    return null;
+  }
+
+  /**
+   * Identifies the name of the resource.
+   *
+   * <p>In the name is {@code null}, it means "all resources" within the scope.
+   */
+  @Value.Default
+  default String element() {
```

Comment:
Maybe we could create a simple abstract class `Element`. Then it will have two child classes:
1.  has only one field: `elementName`. 
2. It has no elements but it denotes this special case: `all resources`.

Reply:
Yes, that's doable. I did not want to create too many classes, but if you find it useful, I'll add that.

Reply:
Added `EntitySelector` for this purpose.

Reply:
I also replaced old usages of `AuthorizedRole` with the new `EntitySelector`